src/main/java/mate/academy/internetshop/dao/jdbc/JdbcOrderDao.java

```diff
@@ -0,0 +1,154 @@
+        }
+    }
+
+    @Override
+    public Optional<Order> get(Long id) {
+        try (Connection con = ConnectionUtil.getConnection()) {
+            String query = "SELECT user_id FROM orders WHERE order_id = ?";
+            PreparedStatement statement = con.prepareStatement(query);
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(getOrder(id, resultSet.getLong("user_id")));
+            }
+            return Optional.empty();
+        } catch (SQLException e) {
+            LOGGER.error(e);
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public List<Order> getAll() {
+        try (Connection con = ConnectionUtil.getConnection()) {
+            String query = "SELECT * FROM orders";
+            Statement statement = con.createStatement();
+            List<Order> orders = new ArrayList<>();
+            if (statement.execute(query)) {
+                ResultSet resultSet = statement.getResultSet();
+                while (resultSet.next()) {
+                    orders.add(getOrder(resultSet.getLong("order_id"),
+                            resultSet.getLong("user_id")));
+                }
+            }
+            return orders;
+        } catch (SQLException e) {
+            LOGGER.error(e);
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public Order update(Order order) {
+        try (Connection con = ConnectionUtil.getConnection()) {
+            String query = "UPDATE orders SET user_id = ? WHERE order_id = ?";
+            PreparedStatement statement = con.prepareStatement(query);
+            statement.setLong(1, order.getUser().getId());
+            statement.setLong(2, order.getId());
+            statement.executeUpdate();
+            query = "DELETE FROM orders_products WHERE order_id = ?";
+            statement = con.prepareStatement(query);
+            statement.setLong(1, order.getId());
+            statement.executeUpdate();
+            saveProducts(order, con);
+            return get(order.getId()).get();
+        } catch (SQLException e) {
+            LOGGER.error(e);
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        try (Connection con = ConnectionUtil.getConnection()) {
+            String query = "DELETE FROM orders WHERE order_id = ?";
+            PreparedStatement statement = con.prepareStatement(query);
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            LOGGER.error(e);
+            throw new RuntimeException(e);
+        }
+    }
+
+    private Order getOrder(Long id, Long userId) throws SQLException {
+        return new Order(id, productDao.getByOrder(id),
+                userDao.get(userId).orElseThrow());
```

Comment:
you have an option to replace User user in ShoppingCart and Order, to Long userId. This way you will be able to remove usage of userDao from others Daos

Reply:
In this case I should find user by Id in GetOrderController, GetUserOrdersController, GetShoppingCartController and CompleteOrderController. If I may I choose to leave it as it is.

Reply:
Yes, and if you think about if you have method where you should find something by user, it's not too hard to modify this method to find something by userId. It's up to you after all)